src/mpid/ch4/shm/ipc/xpmem/xpmem_init.c

```diff
@@ -26,8 +26,16 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *tag_bits)
     /* Try to share entire address space */
     MPIDI_XPMEMI_global.segid = xpmem_make(0, XPMEM_MAXADDR_SIZE, XPMEM_PERMIT_MODE,
                                            MPIDI_XPMEMI_PERMIT_VALUE);
-    /* 64-bit segment ID or failure(-1) */
+#ifdef MPIDI_CH4_SHM_XPMEM_ALLOW_SILENT_FALLBACK
+    if (MPIDI_XPMEMI_global.segid == -1) {
+        /* do not throw an error; instead gracefully disable XPMEM */
+        goto fn_fail;
+    }
+#else
```

Comment:
We need to always exchange the `segid` to check if any other process on the node fails. If any failure occurs, we disable xpmem on the entire node. So we don't `goto fn_fail` here.

Reply:
If you make this change, we need to always release `MPIDI_XPMEMI_global.segmaps` at finalize.

Reply:
No, this is not meant as a fallback for anything other than XPMEM is not available on the node.  That is, if one process fails, all processes fail.